source/extensions/common/wasm/context.cc

```diff
@@ -1703,6 +1708,68 @@ WasmResult Context::grpcCancel(uint32_t token) {
   return WasmResult::Ok;
 }
 
+WasmResult Context::exprCreate(absl::string_view expr, uint32_t* token_ptr) {
+  auto parse_status = google::api::expr::parser::Parse(std::string(expr));
+  if (!parse_status.ok()) {
+    ENVOY_LOG(info, "wasm expr parse {}: {}", log_prefix(), parse_status.status().message());
+    return WasmResult::BadArgument;
+  }
+
+  // lazily create a builder
+  if (!builder_) {
+    google::api::expr::runtime::InterpreterOptions options;
+    builder_ = google::api::expr::runtime::CreateCelExpressionBuilder(options);
+    auto register_status =
+        google::api::expr::runtime::RegisterBuiltinFunctions(builder_->GetRegistry(), options);
+    if (!register_status.ok()) {
+      return WasmResult::InternalFailure;
+    }
+  }
+
+  auto token = next_expr_token_++;
+  for (;;) {
+    if (!expr_.count(token)) {
```

Comment:
token is monotonically increasing. Does the map have a different life time than next_expr_token ?
when will this condition occur?

Reply:
Map is hosted in the root context I think. It's a common pattern in this file to have this token loop.